scope/scope.go

```diff
@@ -376,6 +376,10 @@ func (self *Scope) Log(format string, a ...interface{}) {
 	self.dispatcher.Log(format, a...)
 }
 
+func (self *Scope) Error(format string, a ...interface{}) {
```

Comment:
So this doesnt really help because all the external code uses the Scope interface. We would need to extend the interface and that is already huge - is it worth it? IDK.

Also there should be no leading space after ":"

Reply:
I think it's a little more ergonomic to call scope.Error rather than doing scope.Log("ERROR:...") everywhere you have an error. All calls to scope.Log that are meant to be an error could be refactored to just call scope.Error

Reply:
But why just error? Should we have Warn() or Debug()? Maybe we need a different method LogWithLevel(level, msg)?

Reply:
[Logrus](https://github.com/sirupsen/logrus#level-logging) and a lot of other logging libraries support `log.Debug`,`log.Error`,`log.Info`, etc. I think it makes sense to add methods for each log level.

Reply:
Ok let's add it to the interface here
https://github.com/Velocidex/vfilter/blob/7af3cd80b93459c49cc346fa1894f7c55e3eefe7/types/scope.go#L70